syntax = "proto3";

service OrderService {
  rpc CreateOrder (OrderRequest) returns (OrderResponse);
}

message OrderRequest {
  string product_name = 1;
  float price = 2;
  string payment_gateway = 3;
  string card_brand = 4;
  string bank = 5;
  ShippingAddress shipping_address = 6;
  string customer_email = 7;
}

message OrderResponse {
  string order_id = 1;
  string status = 2;
}

message ShippingAddress {
  string region = 1;
  string address = 2;
}
